cpp/src/io/orc/reader_impl.cu

```diff
@@ -215,6 +218,257 @@ size_t gather_stream_info(const size_t stripe_index,
+      num_rows(calc_num_rows()),
+      num_columns(calc_num_cols()),
+      num_stripes(calc_num_stripes())
+  {
+    // Verify that the input files have the same number of columns,
+    // as well as matching types, compression, and names
+    for (auto const &pfm : per_file_metadata) {
+      CUDF_EXPECTS(per_file_metadata[0].get_num_columns() == pfm.get_num_columns(),
+                   "All sources must have the same number of columns");
+      CUDF_EXPECTS(per_file_metadata[0].ps.compression == pfm.ps.compression,
+                   "All sources must have the same compression type");
+
+      // Check the types, column names, and decimal scale
+      for (size_t i = 0; i < pfm.ff.types.size(); i++) {
+        CUDF_EXPECTS(std::equal(pfm.ff.types[i].fieldNames.begin(),
+                                pfm.ff.types[i].fieldNames.end(),
+                                per_file_metadata[0].ff.types[i].fieldNames.begin()),
+                     "All source column names must be the same");
+        CUDF_EXPECTS(
+          pfm.ff.types[i].scale.value_or(0) == per_file_metadata[0].ff.types[i].scale.value_or(0),
+          "All scale values must be the same");
+        CUDF_EXPECTS(pfm.ff.types[i].kind == per_file_metadata[0].ff.types[i].kind,
+                     "Column types across all input sources must be the same");
+      }
+    }
+  }
+
+  auto const &get_schema(int schema_idx) const { return per_file_metadata[0].ff.types[schema_idx]; }
+
+  auto get_metadata_at_idx(int metadata_idx) const { return &per_file_metadata[metadata_idx]; };
+
+  auto get_col_type(int col_idx) const { return per_file_metadata[0].ff.types[col_idx]; }
+
+  auto get_num_rows() const { return num_rows; }
+
+  auto get_num_cols() const { return per_file_metadata[0].get_num_columns(); }
+
+  auto get_num_stripes() const { return num_stripes; }
+
+  auto get_num_source_files() const { return per_file_metadata.size(); }
+
+  auto const &get_types() const { return per_file_metadata[0].ff.types; }
+
+  int get_row_index_stride() const { return per_file_metadata[0].ff.rowIndexStride; }
+
+  auto get_column_name(const int source_idx, const int column_idx) const
+  {
+    return per_file_metadata[source_idx].get_column_name(column_idx);
+  }
+
+  std::vector<cudf::io::orc::metadata::stripe_source_mapping> select_stripes(
+    std::vector<std::vector<size_type>> const &user_specified_stripes,
+    size_type &row_start,
+    size_type &row_count)
+  {
+    std::vector<cudf::io::orc::metadata::stripe_source_mapping> selected_stripes_mapping;
+
+    if (!user_specified_stripes.empty()) {
+      CUDF_EXPECTS(user_specified_stripes.size() == get_num_source_files(),
+                   "Must specify stripes for each source");
+      // row_start is 0 if stripes are set. If this is not true anymore, then
+      // row_start needs to be subtracted to get the correct row_count
+      CUDF_EXPECTS(row_start == 0, "Start row index should be 0");
```

Comment:
is skiprows not supported with specifying stripes ?

Reply:
Later noticed that this behavior was carry-forwarded from earlier implementation, we can tackle this as a different issue if this is required.